shell/src/main/java/org/apache/accumulo/shell/format/DeleterFormatter.java

```diff
@@ -79,33 +79,36 @@ public String next() {
     Mutation m = new Mutation(key.getRow());
     String entryStr = formatEntry(next, isDoTimestamps());
     boolean delete = force;
-    try {
-      if (!force) {
-        shellState.getReader().flush();
+    if (!force) {
+      try {
+        shellState.getWriter().flush();
+        // this will cause end of file Exception for one of the formatter tests
```

Comment:
Which one? Why? Can it be fixed?

Reply:
This is related to the other comment you mentioned in the `DeleterFormatterTest`. I will have to refresh my memory on it a bit.

Reply:
the failure happens on line 86 where it tries to read the delete prompt.  When `DeleterFormatterTest.testNoConfirmation()` is ran, it hits that line but gets an end of file marker instead.

Reply:
It is possible that Jline3 changed something behind the scenes. On the main branch, line returns null when trying to read the prompt. Currently, it reads the end of file. Not sure what the best approach to fix it yet but I am trying to get it to where it returns null.

Reply:
I added a potential fix for this in my latest commit. It is probably good enough without completely rewriting `DeleterFormatter`